common/repositories/base/base_character_expedition_lockouts_repository.h

```diff
@@ -0,0 +1,362 @@
+ * Copyright (C) 2001-2020 EQEmulator Development Team (https://github.com/EQEmu/Server)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY except by those people which sell it, which
+ * are required to give you total support for your newly bought product;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ *
+ */
+
+/**
+ * This repository was automatically generated and is NOT to be modified directly.
+ * Any repository modifications are meant to be made to
+ * the repository extending the base. Any modifications to base repositories are to
+ * be made by the generator only
+ */
+
+#ifndef EQEMU_BASE_CHARACTER_EXPEDITION_LOCKOUTS_REPOSITORY_H
+#define EQEMU_BASE_CHARACTER_EXPEDITION_LOCKOUTS_REPOSITORY_H
+
+#include "../../database.h"
+#include "../../string_util.h"
+
+class BaseCharacterExpeditionLockoutsRepository {
+public:
+	struct CharacterExpeditionLockouts {
+		int         id;
+		int         character_id;
+		std::string expedition_name;
+		std::string event_name;
+		std::string expire_time;
+		int         duration;
+		std::string from_expedition_uuid;
+	};
+
+	static std::string PrimaryKey()
+	{
+		return std::string("id");
+	}
+
+	static std::vector<std::string> Columns()
+	{
+		return {
+			"id",
+			"character_id",
+			"expedition_name",
+			"event_name",
+			"expire_time",
+			"duration",
+			"from_expedition_uuid",
+		};
+	}
+
+	static std::string ColumnsRaw()
+	{
+		return std::string(implode(", ", Columns()));
+	}
+
+	static std::string InsertColumnsRaw()
+	{
+		std::vector<std::string> insert_columns;
+
+		for (auto &column : Columns()) {
+			if (column == PrimaryKey()) {
+				continue;
+			}
+
+			insert_columns.push_back(column);
+		}
+
+		return std::string(implode(", ", insert_columns));
+	}
+
+	static std::string TableName()
+	{
+		return std::string("character_expedition_lockouts");
+	}
+
+	static std::string BaseSelect()
+	{
+		return fmt::format(
+			"SELECT {} FROM {}",
+			ColumnsRaw(),
+			TableName()
+		);
+	}
+
+	static std::string BaseInsert()
+	{
+		return fmt::format(
+			"INSERT INTO {} ({}) ",
+			TableName(),
+			InsertColumnsRaw()
+		);
+	}
+
+	static CharacterExpeditionLockouts NewEntity()
+	{
+		CharacterExpeditionLockouts entry{};
+
+		entry.id                   = 0;
+		entry.character_id         = 0;
+		entry.expedition_name      = "";
+		entry.event_name           = "";
+		entry.expire_time          = current_timestamp();
```

Comment:
I don't see this method in the source, maybe an artifact from the autogen for DATETIME columns?

Reply:
Not sure where this came from tbh

Reply:
It's coming from the MySQL default value column; that's why. Should still probably change how this outputs a bit. Easy enough to change either way

![image](https://user-images.githubusercontent.com/3319450/107108585-bb7fc880-67fe-11eb-8f28-45a158e611ad.png)

Reply:
```perl
        my $default_value = 0;
        if ($column_default ne "NULL" && $column_default ne "") {
            $column_default =~ s/'/"/g;
            $default_value = $column_default;
        }
        elsif ($column_default eq "''") {
            $default_value = '""';
        }
        elsif ((trim($column_default) eq "" || $column_default eq "NULL") && $column_type =~ /text|varchar/i) {
            $default_value = '""';
        }
```